assembly/src/tokens/stream.rs

```diff
@@ -91,6 +96,11 @@ impl<'a> TokenStream<'a> {
         self.pos
     }
 
+    /// Returns the line number of the current token, capped by the last line of the source.
+    pub fn line_number(&self) -> usize {
+        self.lines[self.pos.min(self.lines.len().saturating_sub(1))]
+    }
+
```

Comment:
I think this is used only for the `EOF` error. If so, it may be better to replace this with something like `num_lines()` method. But we can do this in future PRs as well.

Reply:
Makes sense; fixed

Reply:
Also; added #857